examples/mach_tutorial_wing/analysis.py

```diff
@@ -0,0 +1,397 @@
+# Setup static problem
+# ==============================================================================
+problem = FEAAssembler.createStaticProblem("StructAnalysis")
+
+# Add TACS Functions
+problem.addFunction("mass", functions.StructuralMass)
+
+# Add failure and mass functions for each skin and spar/rib group
+failureGroups = ["l_skin", "u_skin", "spar", "rib"]
+if failureGroups is not None:
+    for group in failureGroups:
+        compIDs = FEAAssembler.selectCompIDs(include=group.upper())
+        problem.addFunction(
+            f"{group}_ksFailure",
+            functions.KSFailure,
+            compIDs=compIDs,
+            safetyFactor=1.5,
+            ksWeight=100.0,
+        )
+        problem.addFunction(
+            f"{group}_mass",
+            functions.StructuralMass,
+            compIDs=compIDs,
+        )
+problem.addFunction("compliance", functions.Compliance)
+
+# Add gravity load
+g = np.zeros(3)
+g[-1] = -9.81
+problem.addInertialLoad(2.5 * g)
+
+# Add simple distributed vertical load over the skins
+totalForce = 0.5 * 2.5 * 55e3 * 9.81
+F = np.zeros(6)
+F[2] = totalForce
+skinCompIDs = FEAAssembler.selectCompIDs("SKIN")
+problem.addLoadToComponents(skinCompIDs, F, averageLoad=True)
+
+# ==============================================================================
+# Setup constraints
+# ==============================================================================
+
+thicknessAdjCon = 2.5e-3  # 2.5mm
+heightAdjCon = 1e-2  # 1cm
+pitchAdjCon = 5e-2  # 5cm
+thickDiffMax = (
+    2.5e-3  # 2.5mm, Max allowable thickness difference between skin and stiffener
+)
+stiffAspectMax = 10.0  # Maximum allowable stiffener aspect ratio (height/thickness)
+stiffAspectMin = 2.0  # Minimum allowable stiffener aspect ratio (height/thickness)
+
+compIDs = {}
+for group in ["SPAR", "U_SKIN", "L_SKIN", "RIB"]:
+    compIDs[group] = FEAAssembler.selectCompIDs(include=group.upper())
+
+constraints = []
+
+# Add adjacency constraints on panel thickness, stiffener thickness and stiffener height to everything but the ribs
+adjCon = FEAAssembler.createAdjacencyConstraint("AdjCon")
+for group in ["SPAR", "U_SKIN", "L_SKIN"]:
+    adjCon.addConstraint(
+        conName=group + "_panelThicknessAdj",
+        compIDs=compIDs[group],
+        lower=-thicknessAdjCon,
+        upper=thicknessAdjCon,
+        dvIndex=2,
+    )
+    adjCon.addConstraint(
+        conName=group + "_stiffenerThicknessAdj",
+        compIDs=compIDs[group],
+        lower=-thicknessAdjCon,
+        upper=thicknessAdjCon,
+        dvIndex=4,
+    )
+    adjCon.addConstraint(
+        conName=group + "_stiffenerHeightAdj",
+        compIDs=compIDs[group],
+        lower=-heightAdjCon,
+        upper=heightAdjCon,
+        dvIndex=3,
+    )
+    adjCon.addConstraint(
+        conName=group + "_stiffenerPitchAdj",
+        compIDs=compIDs[group],
+        lower=-pitchAdjCon,
+        upper=pitchAdjCon,
+        dvIndex=2,
```

Comment:
Is this a typo @A-CGray ? Both the pitch and panel thickness seem to be tagged as dvIndex 2 here